great_expectations/data_context/store/inline_store_backend.py

```diff
@@ -168,56 +162,31 @@ def remove_key(self, key: Tuple[str, ...]) -> None:
 
         self._save_changes()
 
+    def build_key(
+        self,
+        id_: Optional[str] = None,
+        name: Optional[str] = None,
+    ) -> DataContextVariableKey:
+        """Get the store backend specific implementation of the key. id_ included for super class compatibility."""
+        return DataContextVariableKey(
+            resource_name=name,
+        )
+
     def _has_key(self, key: Tuple[str, ...]) -> bool:
-        (
-            resource_type,
-            resource_name,
-        ) = InlineStoreBackend._determine_resource_type_and_name_from_key(key)
-
-        if len(key) == 1:
-            return resource_type in self._data_context.config
-        elif len(key) == 2:
+        resource_name = InlineStoreBackend._determine_resource_name(key)
+        resource_type = self._resource_type
+
+        if resource_name is not None:
             res: dict = self._data_context.config.get(resource_type) or {}
             return resource_name in res
 
-        return False
-
-    def _validate_key(self, key: Tuple[str, ...]) -> None:
-        (
-            resource_type,
-            _,
-        ) = InlineStoreBackend._determine_resource_type_and_name_from_key(key)
-
-        if not isinstance(
-            resource_type, DataContextVariableSchema
-        ) and not DataContextVariableSchema.has_value(resource_type):
-            raise TypeError(
-                f"Keys in {self.__class__.__name__} must adhere to the schema defined by {DataContextVariableSchema.__name__}; invalid value ({resource_type}) of type {type(resource_type)} found"
-            )
+        return resource_type in self._data_context.config
```

Comment:
Instead of using tuple length, we can use the presence/absence of a `resource_name` for a bit better of a control flow.

Reply:
A key would be ideal here but will have to come in a subsequent PR